src/openforms/contrib/haal_centraal/tests/test_brp_clients.py

```diff
@@ -549,6 +561,160 @@ def test_invalid_version_raises_error(self):
         with patch(
             "openforms.contrib.haal_centraal.clients.HaalCentraalConfig.get_solo",
             return_value=config,
+        ), patch(
+            "openforms.contrib.haal_centraal.clients.GlobalConfiguration.get_solo",
+            return_value=GlobalConfiguration(),
         ):
             with self.assertRaises(RuntimeError):
-                get_brp_client()
+                get_brp_client(SubmissionFactory.build())
+
+
+class HaalCentraalXHeadersTests(TestCase):
+    def setUp(self):
+        super().setUp()
+
+        # set up patcher for the configuration
+        config = HaalCentraalConfig(
+            brp_personen_service=ServiceFactory.build(
+                api_root="https://personen/api/",
+                oas="https://this.is.ignored",
+            ),
+            brp_personen_version=BRPVersions.v20,
+            default_brp_personen_purpose_limitation_header_value="BRPACT-AanschrijvenZakelijkGerechtigde",
+            default_brp_personen_processing_header_value="Financiële administratie@Correspondentie factuur",
+        )
+        config_patcher = patch(
+            "openforms.contrib.haal_centraal.clients.HaalCentraalConfig.get_solo",
+            return_value=config,
+        )
+        config_patcher.start()
+        self.addCleanup(config_patcher.stop)
+
+        self.requests_mock = requests_mock.Mocker()
+        self.requests_mock.start()
+        self.addCleanup(self.requests_mock.stop)
+
+    @patch(
+        "openforms.contrib.haal_centraal.clients.GlobalConfiguration.get_solo",
+        return_value=GlobalConfiguration(),
+    )
+    def test_uses_default_headers(self, m):
+        self.requests_mock.post(
+            "https://personen/api/irrelevant",
+            status_code=200,
+        )
```

Comment:
You can also just blanket mock any request (and then maybe generate a URL using hypothesis to show this applies to any URL/method)

```suggestion
        self.requests_mock.register_uri(requests_mock.ANY, requests_mock.ANY, status_code=200)
```
doesn't necessarily need to be changed, just FYI